draft-ietf-spice-sd-cwt.md

```diff
@@ -409,22 +409,132 @@ Since the unprotected header of the included SD-CWT is covered by the signature
 
 # Differences from the CBOR Web Token Specification {#cwt-diffs}
 
-The CBOR Web Token Specification (Section 1.1 of {{!RFC8392}}), uses text strings, negative integers, and unsigned integers as map keys.
-This specification also allows the CBOR simple value registered in this specification in {{simple59}}, and CBOR tagged integers and text strings as map keys.
-As in CWTs, CBOR maps used in an SD-CWT or SD-KBT also cannot have duplicate keys.
-(An integer or text string map key is a distinct key from a tagged map key that wraps the corresponding integer or text string value).
+## Definite Length CBOR Required
+
+Encoders of SD-CWT and SD-KBT MUST NOT send indefinite length CBOR.
+Decoders of SD-CWT and SD-KBT MUST reject any SD-CWT or SD-KBT received containing indefinite length CBOR.
+
+
+## Finite values for standard date claims
+
+The standard CWT claims `exp`, `nbf`, and `iat` MUST be finite numbers.
+For the avoidance of doubt, not a number (NaN) values and positive and negative infinity are not acceptable in those claims.
+
+> In {{!RFC8392}}, these three claims are of type `NumericDate`.
+Section 2 of the same spec refers to `NumericDate` as a JWT `NumericDate`, "except that it is represented as [an untagged] CBOR numeric date (from {{Section 2.4.1 of ?RFC7049}}) instead of a JSON number".
+In CBOR, a NumericDate can be represented as an unsigned integer, a negative integer, or a floating point value.
+CBOR (both {{?RFC7049}} and {{!RFC8949}}) refers to floating-point values to include NaNs, and floating-point numbers that include finite and infinite numbers.
+Neither JSON {{?RFC8259}} nor JWT {{?RFC7519}} can represent infinite values.
+
+
+## Allowed types of CBOR map keys
+
+According to Section 1.1 of the CBOR Web Token Specification {{!RFC8392}}, "CBOR uses text strings, negative integers, and unsigned integers as map keys."
+Section 1.5 of CBOR Object Signing and Encryption (COSE): Structures and Process {{!RFC9052}} states: "In COSE, we use text strings, negative integers, and unsigned integers as map keys."
+While a CBOR map key can contain any CBOR type, this statement implies that CWT map keys only contain those types.
+
+An SD-CWT payload is typically its COSE payload.
+{{?RFC9597}} also defines the `CWT Claims` COSE Header Parameter (value 15) that can appear in the protected headers; if it exists, it contains a CBOR map with additional claims that are treated as if they were in the SD-CWT payload.
+Both of these maps are described as SD-CWT Claims Maps.
+
+> Note that `CWT Claims` is a separate CBOR map from the COSE payload and can contain the same Claim Keys as the COSE payload CBOR map.
+The same claim key could be present in both maps and could have different values in those maps.
+Neither, one, or both could be redacted, and if both are redacted they would have different disclosures, salts, and Blinded Claim Hashes.
```

Comment:
https://www.rfc-editor.org/rfc/rfc9597.html#name-representation says:
> In cases where CWT claims are present both in the payload and the header of a CWT, an application receiving such a structure MUST verify that their values are identical, unless the application defines other specific processing rules for these claims.

We seem to be looser than RFC 9597 in that we're not repeating that text about verifying that claims appearing twice MUST be identical.  I'd advocate repeating the sentence above in SD-CWT.